api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -123,6 +123,8 @@ message Cluster {
   // only perform a lookup for addresses in the IPv6 family. If AUTO is
```

Comment:
I agree this would be optimal but I don't think it's worth the API deprecation. You could I suppose also add V6_PREFERRED and map it to the existing AUTO? Not sure.

Reply:
Makes sense. I was thinking just of the enum in C++ but of course this is part of the API so the overhead of the change is probably not worth it.

Reply:
I think adding V6_PREFERRED and just aliasing it to AUTO would be a reasonable compromise. I agree the lack of symmetry is confusing, but not sure it's worth it. Will leave it for you and @junr03 to decide.

Reply:
Yeah I thought about this too, as AUTO really _is_ V6_PREFERRED. I think AUTO is actually misleading if the operator doesn't read the docs. 

@htuch what is the kosher way to deprecate an enum? Create a new value that we actually want alias (like matt suggests) and in the long distant future not include the value we no longer want?

Reply:
Since AUTO is the default value, I'm not sure it really can be deprecated TBH, though I'm not positive. You could try it and see what happens, though I'm not sure it's worth it. I would probably just document very clearly that it is V6_PREFERRED effectively and then we could rename it in a "future version."

Reply:
Yeah, I don't think it's worth the hassle for this one. I also had the same reaction TBH, and really, the right thing is to make the default value to always have "client specific semantics". That way, you would always have a `V6_PREFERRED` as a separate value as well as Envoy's default.

Since we've advertised `AUTO` as having very specific behavior, I don't think it is legitimate to change this. The cleanest thing to do would be to add `V6_PREFERRED` as an alias and retain `AUTO` existing behavior.

CC @envoyproxy/api-shepherds for further discussion.

Reply:
I would probably just skip any deprecation on this. I don't think it's worth it. I would just update the docs to make it very clear that AUTO is basically V6_PREFERRED.